src/utils/groupByUniqName.js

```diff
@@ -0,0 +1,17 @@
+import { isEmpty, chain, has } from 'lodash'
+
+export const groupByUniqName = (value, field = '') => {
+  const makeUniq = items => {
+    return chain(items)
+      .filter(item => has(item, field + '.name') || has(item, 'name'))
+      .uniqBy(field + '.name')
+      .value()
+  }
+  if (!isEmpty(value.data)) {
+    return {
+      data: makeUniq(value.data)
+    }
+  } else {
+    return makeUniq(value)
+  }
```

Comment:
When I see:
```js
if (condition) {
  return value1
} else {
  return value2
}
```
meaning an `if` that only has a `return` inside and nothing more, then `else` that only has a `return` inside and nothing more — it means that whether or not the condition passes a `return` will happen, only the value returned could be either one of two possible values, depending on the condition.
Then I think about this simplification:
```js
return condition ? a : b
```
Note that this will not be possible if there was another statement, for example:
```js
if (condition) {
  return a
} else {
  doSomething // this
  return b
}
```
Here, because of `doSomething`, we can't simplify in the same way we did above.